src/snowflake/snowpark/session.py

```diff
@@ -830,19 +857,39 @@ def add_requirements(self, file_path: str) -> None:
             and the Snowflake server.
         """
         packages = []
-        with open(file_path) as f:
-            for line in f:
-                package = line.rstrip()
-                if package:
-                    packages.append(package)
-        self.add_packages(packages)
+        if file_path.endswith(".txt"):
+            with open(file_path) as f:
+                for line in f:
+                    package = line.rstrip()
+                    if package:
+                        packages.append(package)
+        elif file_path.endswith(".yml") or file_path.endswith(".yaml"):
+            with open(file_path) as f:
+                try:
+                    environment_data = yaml.safe_load(f)
+                    dependencies = environment_data.get("dependencies", [])
+                    is_python_dependency = (
```

Comment:
We ignore Python itself for now; it will be considered in the next iteration, wherein the RUNTIME_VERSION parameter while creating procedures can be modified.